app/grandchallenge/reader_studies/forms.py

```diff
@@ -593,21 +616,31 @@ def __init__(self, *args, pk, interface, user, **kwargs):
             selected_interface = ComponentInterface.objects.get(
                 pk=data["interface"]
             )
-        ds = DisplaySet.objects.get(pk=pk)
+        if pk is not None:
+            ds = DisplaySet.objects.get(pk=pk)
+            qs = ComponentInterface.objects.exclude(
+                slug__in=ds.reader_study.values_for_interfaces.keys()
+            )
+            attrs = {
+                "hx-get": reverse_lazy(
+                    "reader-studies:display-set-add-interface",
+                    kwargs={"pk": pk},
+                ),
+                "hx-target": f"#ds-content-{pk}",
+            }
+        else:
+            qs = ComponentInterface.objects.all()
```

Comment:
Right now, a user could try to add 2 generic medical images. The form would not validate - good - but the error they get is not clear: "Correct the errors below", but there is no error displayed further below to clarify that the interfaces have to be unique. I think either we need to narrow down the options here or we need a clearer error message for this case.

Reply:
Actually, they don't get an error. The display set gets created without the duplicate interfaces.

Reply:
Nice catch, I'll have a look at limiting the options. Did you manage to figure out what the error was the first time? If not, I'll try to reproduce it.

Reply:
So, the first time around the error was `Correct the errors below` which you get if the view returns a server error (caused by my breakpoint).

Reply:
Ah yes, that message shows on all error response statuses. I;m not sure if we should just add it on 400 status? But in that case, what do we do on other status codes?

Reply:
Ok, I changed the form such that interfaces can only be included once. I also added a delete button for additional interface forms and changed the error message for non 400 status codes.